Fix MSVC build for Python 3.6

smessmer opened the pull request:
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#12878 Fix MSVC build for Python 3.6**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D10467239/)

Python 3.6 headers define their own ssize_t, which clashes with our definition.
Luckily, they also define a `HAVE_SSIZE_T` macro we can use to check for this case.

Differential Revision: [D10467239](https://our.internmc.facebook.com/intern/diff/D10467239/)

yf225 commented:
It's weird that our current Windows CI didn't catch this problem. @peterjc123 Do you have any insights?

ezyang commented:
We are building with Python 3.7. Did 3.7 get rid of this macro?

peterjc123 commented:
We did't set the Python version explicitly so it is actually building for Python 3.7.

smessmer committed:
Fix MSVC build for Python 3.6 (#12878)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/12878

Python 3.6 headers define their own ssize_t, which clashes with our definition.
Luckily, they also define a `HAVE_SSIZE_T` macro we can use to check for this case.

Reviewed By: ezyang

Differential Revision: D10467239

fbshipit-source-id: bf9397f9fbbc10b6fb13c500b616c83c72480598
